OpenSim/Simulation/Model/ModelComponent.h

```diff
@@ -108,6 +110,45 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(ModelComponent, Component);
      */
     Model& updModel();
 
+    /** Perform any computations that must occur before ModelComponent::scale()
+        is invoked on all ModelComponents in the Model. For example, a
+        GeometryPath must calculate and store its path length in the current
+        body position before scaling so that an owning Muscle can use this
+        information to update the properties of the muscle after scaling. This
+        method is virtual and may be implemented by any subclass of
+        ModelComponent, but all implementations must begin with a call to
+        `Super::preScale()` to execute the parent class methods before the child
+        class method. The base class implementation in ModelComponent does
+        nothing.
+        @see scale()
+        @see postScale() */
+    virtual void preScale(const SimTK::State& s, const ScaleSet& scaleSet) {};
+
+    /** Scale the ModelComponent. This method is virtual and may be implemented
+        by any subclass of ModelComponent, but all implementations must begin
+        with a call to `Super::scale()` to execute the parent class methods
+        before the child class method. The base class implementation in
+        ModelComponent does nothing.
+        @see preScale()
+        @see postScale() */
+    virtual void scale(const SimTK::State& s, const ScaleSet& scaleSet) {};
```

Comment:
Is it your plan that is will no longer be virtual and `extendScale()` will be virtual? Just double checking.

Reply:
Yes, that's the plan: remove default arguments from `Model::scale()`; rename current virtual methods to `extendPreScale()`, `extendScale()`, and `extendPostScale()`; and create non-virtual `preScale()`, etc. methods on ModelComponent, per discussion.

Reply:
Oh, actually I don't think it's a good idea to remove default arguments from the existing `Model::scale()` because then the user would need to pass in `-1.0` (or any number less than zero) for `finalMass` to get the same behavior. I'll work out something else.

Reply:
I'll change `bool Model::scale(State&, const ScaleSet&, double finalMass = -1.0, bool preserveMassDist = false)` to `bool Model::scale(State&, const ScaleSet&, bool preserveMassDist, double finalMass = -1.0)`. The user will now need to specify whether they want to preserve the mass distribution, but can still leave `finalMass` unspecified (which I think is better than entering `-1` for a mass). I'll make a note of this change in CHANGELOG.md.